Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs

```diff
@@ -1392,6 +1392,30 @@ private bool ShouldIgnoreControlStateChange(ref TriggerState trigger, int action
                     trigger.bindingIndex = bindingWithHighestActuation;
                     trigger.magnitude = highestActuationLevel;
 
+                    // If we're switching to a different binding, we may also have to switch to a
+                    // different stack of interactions.
+                    if (actionState->bindingIndex != bindingWithHighestActuation)
+                    {
+                        // If there's an interaction currently driving the action, reset it.
+                        if (actionState->interactionIndex != kInvalidIndex)
+                            ResetInteractionState(trigger.mapIndex, actionState->bindingIndex, actionState->interactionIndex);
+
+                        // If there's an interaction in progress on the new binding, let
+                        // it drive the action.
+                        var bindingState = &bindingStates[bindingWithHighestActuation];
+                        var interactionCount = bindingState->interactionCount;
+                        var interactionStartIndex = bindingState->interactionStartIndex;
+                        for (var i = 0; i < interactionCount; ++i)
+                        {
+                            if (!interactionStates[interactionStartIndex + i].phase.IsInProgress())
```

Comment:
Since we compute interactionStartIndex + i 3 times within this for maybe compute once as a local and reuse?